docs/eventing/sources/rabbitmq-source/README.md

```diff
@@ -0,0 +1,219 @@
+     ```
+
+1. Apply the YAML file by running the command:
+
+    ```bash
+    kubectl apply -f <filename>.yaml
+    ```
+    Where `<filename>` is the name of the file you created in the previous step.
+
+    Example output:
+    ```{ .bash .no-copy }
+    service.serving.knative.dev/event-display created
+    ```
+
+1. Ensure that the Service Pod is running, by running the command:
+
+    ```bash
+    kubectl get pods
+    ```
+
+    The Pod name is prefixed with `event-display`:
+    ```{ .bash .no-copy }
+    NAME                                            READY     STATUS    RESTARTS   AGE
+    event-display-00001-deployment-5d5df6c7-gv2j4   2/2       Running   0          72s
+    ```
+
+## Create a RabbitMQ Source object
+
+1. Create a YAML file using the following template:
+
+    ```yaml
+    apiVersion: sources.knative.dev/v1alpha1
+    kind: RabbitmqSource
+    metadata:
+      name: <source-name>
+    spec:
+      rabbitmqClusterReference:
+        # Configure name if a local cluster is being used.
+        name: <cluster-name>
+        # Configure connectionSecret if an external RabbitMQ cluster is being used.
+        connectionSecret:
+          name: rabbitmq-secret-credentials
+      rabbitmqResourcesConfig:
+        parallelism: 10
+        exchangeName: "eventing-rabbitmq-source"
+        queueName: "eventing-rabbitmq-source"
+      delivery:
+        retry: 5
+        backoffPolicy: "linear"
+        backoffDelay: "PT1S"
+      sink:
+        ref:
+          apiVersion: serving.knative.dev/v1
+          kind: Service
+          name: event-display
+    ```
+    Where:
+
+    - `<source-name>` is the name you want for your RabbitMQ Source object.
+    - `<cluster-name>` is the name of the RabbitMQ cluster you created earlier.
+
+    !!! note
+        You cannot set `name` and `connectionSecret` at the same time.
+
+1. Apply the YAML file by running the command:
+
+    ```bash
+    kubectl apply -f <filename>
+    ```
+    Where `<filename>` is the name of the file you created in the previous step.
+
+### Verify
+
+Check the event-display service to see if it is receiving events.
+It might take a while for the Source to start sending events to the Sink.
+
+```sh
+  kubectl -l='serving.knative.dev/service=event-display' logs -c user-container
+  ☁️  cloudevents.Event
+  Context Attributes,
+    specversion: 1.0
+    type: dev.knative.rabbitmq.event
+    source: /apis/v1/namespaces/default/rabbitmqsources/<source-name>
+    subject: f147099d-c64d-41f7-b8eb-a2e53b228349
+    id: f147099d-c64d-41f7-b8eb-a2e53b228349
+    time: 2021-12-16T20:11:39.052276498Z
+    datacontenttype: application/json
+  Data,
+    {
+      ...
+      Random Data
+      ...
+    }
+```
+
+Congratulations! Your new RabbitMQ Source is working!
+
+### Cleanup
+
+```sh
+kubectl delete -f <source-yaml-filename> <service-yaml-filename> <secret-yaml-filename> <cluster-yaml-filename>
+```
```

Comment:
explain exactly what's being removed here / why run the command

is it worth creating a namespace just for RabbitMQ and then just deleting everything in the ns at the end? Idk if that's possible but it might make cleanup easier generally